### ISSUE TITLE
mask generated by Embedding and Masking have different type in tf

### ISSUE THREAD

**Issue by udibr** [fchollet/keras#2639]
In Theano they both give the same type but in Tensorflow the mask generated by Embedding is of type bool and the mask generated by Masking is uint8 if you latter try to concatenate the masks you get an error in tensorflow unless you first cast them to uint8 (in theano there is no bool so if you want a universal code it is better to cast to uint8 than to bool)
